scripts/zones/Maze_of_Shakhrami/mobs/Argus.lua

```diff
@@ -0,0 +1,23 @@
+-----------------------------------
+-- Area: Maze of Shakhrami
+--  MOB: Argus (NM)
+-----------------------------------
+
+require("scripts/zones/Maze_of_Shakhrami/MobIDs");
+
+-----------------------------------
+-- onMobDeath
+-----------------------------------
+
+function onMobDeath(mob,killer,ally)
+    local whichNM = math.random(0,19);
+
+    if (whichNM < 10) then
+        SetRespawnTime(Argus, 64800, 108000); -- 18-30 hours
+        DeterMob(Leech_King, true);
+    else
+        SetRespawnTime(Leech_King, 64800, 108000); -- 18-30 hours
+        DeterMob(Argus, true);
```

Comment:
Tested and not neccessary, as CLuaBaseEntity::setRespawnTime per default sets

PMob->m_AllowRespawn = true;

I know it looks a bit short/incomplete but I tested alot and checked the underlying functions, so it should work as expected